src/pal/tools/gen-buildsys-gcc.sh

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
```

Comment:
Please consider a POSIX shell variant of this script to avoid dependency on bash: https://gist.github.com/am11/41aa89781164673d69019caf597ea580, few lines were modified and also fixed some warnings reported by https://www.shellcheck.net/.

Reply:
thanks, incorporated your version.

Reply:
Thanks. I realized that we could use one of the two styles consistently for conditions:
`if test ...; then` or `if [ ... ]; then`
gist was using both, I have updated it to use latter one in [new revision](https://gist.github.com/am11/41aa89781164673d69019caf597ea580/revisions), and changed OS condition `if test cond1 -o cond2 -o..` line to `if [cond1] || [cond2] ||..` pattern per scriptchecker warning.